### PR TITLE
[IMP] mail: turn "RtcSession/isConnected" into field

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2734101

### CODE REVIEW

File: addons/mail/static/src/models/rtc/rtc.js [hunk 1]
```diff
@@ -759,7 +758,9 @@ registerModel({
             delete this._peerConnections[token];
             browser.clearTimeout(this._fallBackTimeouts[token]);
             delete this._fallBackTimeouts[token];
-            this._outGoingCallTokens.delete(token);
+            const newOutGoingCallTokens = { ...this.outGoingCallTokens };
+            delete newOutGoingCallTokens[token];
+            this.update({ outGoingCallTokens: newOutGoingCallTokens });
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:763]: make new model instead?

File: addons/mail/static/src/models/rtc_outgoing_call/rtc_outgoing_call.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+
+registerModel({
+    name: 'RtcOutgoingCall',
+    identifyingFields: ['rtc', 'token'],
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc_outgoing_call/rtc_outgoing_call.js:8]: As `rtc` is a messaging singleton, I don't think it's necessary to have it as identifying. It's like having `and messaging` as identifying on 100% of the models, sure that works, but that would be overkill.

File: addons/mail/static/src/models/rtc/rtc.js [hunk 3]
```diff
@@ -1278,6 +1272,18 @@ registerModel({
...
+         * which is used when attempting to recover a failed peer connection by
+         * inverting the call direction.
+         *
+         * Key: token
+         * Value: truthy when set
+         */
+        rtcOutgoingCalls: many('RtcOutgoingCall', {
+            inverse: 'rtc',
+            isCausal: true,
+        }),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc/rtc.js:1286]: Looks likes "is outgoing" should be flag on a more generic "call" model instead?

Comment (seb-odoo) [reply addons/mail/static/src/models/rtc/rtc.js:1286]: Or more generally speaking, the first model to make in RTC should be `peer`, with `token` as identifying?
It looks all other models that will also use `token` as identifying should be just be related from peer?

Comment (alexkuhn) [reply addons/mail/static/src/models/rtc/rtc.js]: Done by turning all of this into `RtcSession/isOutgoing`. `token` is actually equivalent to `RtcSession/id` (= id of `RtcSession`), so all is on `RtcSession` when identified by `token`/`peer`.

File: addons/mail/static/src/models/rtc_session/rtc_session.js [hunk 4]
```diff
@@ -355,6 +355,14 @@ registerModel({
             compute: '_computeIsMute',
             default: false,
         }),
+        /**
+         * States whether this RTC session is outgoing.
+         * Useful when attempting to recover a failed peer connection by
+         * inverting the call direction.
+         */
+        isOutgoing: attr({
```
Comment (seb-odoo) [inline addons/mail/static/src/models/rtc_session/rtc_session.js:363]: `isConnected`?  or just existence of browser peer?

**Comment by seb-odoo** [comment]
conflict

File: addons/mail/static/src/models/rtc.js [hunk 5]
```diff
@@ -765,7 +762,7 @@ registerModel({
                 peerConnection.close();
             }
             delete this._peerConnections[sessionId];
-            this._outGoingCallTokens.delete(sessionId);
+            this.messaging.models['RtcSession'].insert({ id: sessionId, isConnected: false });
```
Comment (phenix-factory) [inline addons/mail/static/src/models/rtc.js:765]: Let's be consistent and use multi-line object here.

**Review by phenix-factory — APPROVED** [review]
LGTM

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**